### DISCUSSION TITLE
Imu estimation of 6DoF

### DISCUSSION THREAD

**Discussion by Olofforsberg** [rodralez/NaveGo#54] [Ideas]
Hello

Thanks you guys for NaveGo! It's really nice work.

For an application I am investigating I would like to estimate a 6DoF pose (x,y,z,roll,pitch,yaw), with only one or more 9DoF IMU. I know that there will be drift, and I want to have a feeling of how much and if the tech can realise the application. 

So I'm wondering if I can use NaveGO to evaluate the performance of such system. I had a look into the examples and see that this library seem to be made for IMU+GNSS estimation. Still could I use it for my purpose? Could you give me some pointers for how to set up an example file. If I get it right I would contribute with the example, (if you want it). 

What I have tried before finding Navgo is to use a complementary filter to estimate, roll, pitch yaw and after this, I remove gravity and then integrate the acceleration to get velocity and position. For real time data holding the IMU stationary, I saw drift around 0.5m/s in x,y plane. Would you propose a different approach?

Best Regards
Olof

**Comment by BrunoOtavio98** [comment]
Hi @Olofforsberg look at ins_gnss.m code starting at  "  gdx =  find (gnss.t >= (imu.t(i) - gnss.eps) & gnss.t < (imu.t(i) + gnss.eps));" and below, thats the point where the update phase of kalman is made, but since you want to "turn off" gps part, where correction is made, i think that what you have to do is comment this part of the code and other parts where gps data is used or pass a imu timestamp that never sincronizes with gps timestamp, in this way, kalman update will never be executed. But i think thats a better idea to use some AHRS sensor fusion like madgwick or mahony, thats used to estimate roll, pitch and yaw. I dont think that you will have useful information estimating pose just with imu. I have opened a discussion  https://github.com/rodralez/NaveGo/discussions/53, take a look at my IMU dataset. Dont forget to run allan variance to get the imu error model.

**Comment by rodralez** [comment]
Dear Olof,

As Bruno told you, what you need to code is a AHRS. This system uses accrs, gyros and sometimes magnetometer to estimate the attitude. You can use some functions from NaveGo, mostly from the ins folder, but eventually you will need to implement other functionalities.

Best regards.
